crypto/evp/m_sigver.c

```diff
@@ -191,6 +194,14 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
              */
             ctx->digest = ctx->reqdigest = ctx->fetched_digest =
                 EVP_MD_fetch(locpctx->libctx, mdname, props);
+            /* legacy code support for engines */
+            if (ctx->digest == NULL) {
+                ctx->digest = ctx->reqdigest = EVP_get_digestbyname(mdname);
+                if (ctx->digest != NULL)
+                    ERR_pop_to_mark();
```

Comment:
The flow is still not right..
What about something like this.

``` 
           ERR_set_mark();
            ctx->fetched_digest = EVP_MD_fetch(locpctx->libctx, mdname, props);
            /* legacy code support for engines */
            if (ctx->fetched_digest != NULL)
               ctx->digest = ctx->fetched_digest;
            else
               ctx->digest = EVP_get_digestbyname(mdname);
             ctx->reqdigest = ctx->digest;
             if (ctx->digest == NULL)
                  (void)ERR_clear_last_mark();
                   ERR_raise(ERR_LIB_EVP, EVP_R_INITIALIZATION_ERROR);
                   goto err;
            }
            (void)ERR_pop_to_mark();
```

Reply:
@slontis thank you for your proposal! Force-pushed.